python/ray/data/tests/test_dataset_logger.py

```diff
@@ -0,0 +1,47 @@
+import pytest
+from ray.tests.conftest import *  # noqa
+from unittest.mock import patch, mock_open
+
+import os
+import re
+import logging
+
+from datetime import datetime
+import pytz
+from freezegun import freeze_time
+
+import ray
+from ray.data._internal.dataset_logger import DatasetLogger
+
+
+@freeze_time("2022-11-01 01:23:45")
+def test_dataset_logger(ray_start_regular_shared):
+    curr_ts = datetime.now(pytz.timezone("US/Pacific"))
+    log_name, msg = "test_name", "test message 1234"
+    with patch("logging.open", mock_open(), create=True) as open_mock:
+        logger = DatasetLogger(log_name).logger
+        logger.info(msg)
+
+    session_dir = ray._private.worker._global_node.get_session_dir_path()
```

Comment:
to make the unit test work, could you change this test from `ray_start_regular_shared` to `shutdown_only`, and call `ray.init()` before this? This will initiate a Ray cluster.

Reply:
@c21 thanks for the suggestion, tried this but still seeing this error in the [build tests](https://buildkite.com/ray-project/oss-ci-build-pr/builds/6247#0184a6a8-d26e-48ea-a0ea-e7e4fef1b5d7) (passes when i run test locally):

```


shutdown_only = None
--
  |  
  | def test_dataset_logger(shutdown_only):
  | ray.init()
  | log_name, msg = "test_name", "test message 1234"
  | with patch("logging.open", mock_open(), create=True) as open_mock:
  |  
  | logger = DatasetLogger(log_name).logger
  | logger.info(msg)
  |  
  | session_dir = ray._private.worker._global_node.get_session_dir_path()
  | log_file_path = os.path.join(session_dir, DatasetLogger.DEFAULT_DATASET_LOG_PATH)
  | open_mock.assert_called_with(log_file_path, "a", encoding=None)
  | open_mock.return_value.write.assert_called_once()
  |  
  | # Parse each component of emitted log row
  | raw_logged_msg = open_mock.return_value.write.call_args.args[0].strip()
  | >       log_header, logged_msg = raw_logged_msg.split(" -- ")
  | E       ValueError: not enough values to unpack (expected 2, got 1)
  |  
  | python/ray/data/tests/test_dataset_logger.py:30: ValueError
```

Reply:
resolved by reading the logs directly from the log file instead.